rpc/src/v1/impls/eth_pubsub.rs

```diff
@@ -50,6 +52,70 @@ pub struct EthPubSubClient<C> {
 	heads_subscribers: Arc<RwLock<Subscribers<Client>>>,
 	logs_subscribers: Arc<RwLock<Subscribers<(Client, EthFilter)>>>,
 	transactions_subscribers: Arc<RwLock<Subscribers<Client>>>,
+	sync_subscribers: Arc<RwLock<Subscribers<Client>>>,
+}
+
+impl<C> EthPubSubClient<C>
+	where
+		C: 'static + BlockChainClient
+{
+	/// adds a sync notification channel to the pubsub client
+	pub fn add_sync_notifier(&mut self, receiver: Notification<SyncState>, sync: Arc<SyncProvider>) {
+		let handler = self.handler.clone();
+		let client = self.handler.client.clone() as Arc<BlockChainClient>;
+
+		self.handler.executor.spawn(
+			receiver.for_each(move |state| {
+				let queue_info = client.queue_info();
+				let status = sync.status();
+
+				let is_syncing_state = match state { SyncState::Idle | SyncState::NewBlocks => false, _ => true };
+				let is_verifying = queue_info.unverified_queue_size + queue_info.verified_queue_size > 3;
+
+				let highest_block = U64::from(status.highest_block_number.unwrap_or(status.start_block_number));
+
+				let sync_state = pubsub::PubSubSyncStatus {
+					is_syncing: is_verifying || is_syncing_state,
+					starting_block: status.start_block_number.into(),
+					current_block: client.chain_info().best_block_number.into(),
```

Comment:
So this is being triggered only when `sync` changes status right? Then IMHO returning this additional data here doesn't make sense, cause it's just misleading for users - that stuff won't change at all, since we only notify about sync/notsync.

Either we need to trigger this notification more often (after every imported block for instance), but we discussed with @axelchalon that it might be overwhelming for subscribers.

I'd go for `true/false` returned from subscription and then the clients can poll at their discretion for the proper sync status. Note that we also have subscription for `newHeads` so the two subscriptions combined give you pretty good understanding what happens with the node.

Calling `queue_info` an `chain_info` from client seems pretty heavy, I'd prefer to not do it at all.
Regarding `sync.status()` - why can't we just pass all the details in the `state` already? It's just redundant call that requires some locks to be acquired while we can provide all necessary data with every event.